app/src/main/java/com/bnyro/recorder/ui/models/PlayerModel.kt

```diff
@@ -1,85 +1,64 @@
 package com.bnyro.recorder.ui.models
 
 import android.content.Context
-import android.media.MediaMetadataRetriever
 import android.media.MediaPlayer
-import android.provider.OpenableColumns
 import android.util.Log
 import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableStateListOf
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.setValue
-import androidx.compose.runtime.toMutableStateList
 import androidx.documentfile.provider.DocumentFile
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelProvider.AndroidViewModelFactory.Companion.APPLICATION_KEY
 import androidx.lifecycle.viewModelScope
-import com.bnyro.recorder.enums.RecorderType
+import androidx.lifecycle.viewmodel.initializer
+import androidx.lifecycle.viewmodel.viewModelFactory
+import com.bnyro.recorder.App
 import com.bnyro.recorder.enums.SortOrder
 import com.bnyro.recorder.obj.RecordingItemData
+import com.bnyro.recorder.util.FileRepository
 import com.bnyro.recorder.util.PlayerHelper
-import com.bnyro.recorder.util.StorageHelper
 import java.io.IOException
-import kotlin.math.absoluteValue
-import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
 
-class PlayerModel : ViewModel() {
+class PlayerModel(private val fileRepository: FileRepository) : ViewModel() {
     var isPlaying by mutableStateOf(false)
     var player by mutableStateOf<MediaPlayer?>(null)
     var currentPlayingFile by mutableStateOf<DocumentFile?>(null)
 
-    var files = mutableStateListOf<DocumentFile>()
+    var selectedFiles by mutableStateOf(listOf<RecordingItemData>())
 
-    val recordingItems = mutableStateListOf<RecordingItemData>()
+    private var sortOrder = SortOrder.DEFAULT
 
-    fun loadFiles(context: Context) {
+    var audioRecordingItems by mutableStateOf(listOf<RecordingItemData>())
+    var screenRecordingItems by mutableStateOf(listOf<RecordingItemData>())
+
+    init {
+        loadFiles()
+    }
+
+    fun loadFiles() {
         viewModelScope.launch {
```

Comment:
```suggestion
        viewModelScope.launch(Dispatchers.IO) {
```

Reply:
Are you sure about this one?
As far as I know, the MutableState only works in UI threads

Reply:
NVM it works fine regardless.

Reply:
should I do it like this?

```kotlin
 fun loadFiles() {
        viewModelScope.launch(Dispatchers.IO) {
            val audioItems = fileRepository.getAudioRecordingItems(sortOrder)
            val videoItems = fileRepository.getVideoRecordingItems(sortOrder)
            withContext(Dispatchers.Main) {
                audioRecordingItems = audioItems
                screenRecordingItems = videoItems
            }
        }
    }
```

Reply:
Afaik there's no need to update the MutableState vars on the main thread, I only would do that if using the main thread was mandatory for accessing mutable states.

Reply:
> In Kotlin ViewModel or any other Android architectural component, you should avoid updating mutable state variables on background threads directly. When using Dispatchers.IO in a ViewModel, it's important to be mindful of the threading rules to ensure that mutable state updates happen on the main thread.

Reply:
fair, then your suggestion seems to be the right one

Reply:
I think this is much better

```kotlin
override suspend fun getAudioRecordingItems(sortOrder: SortOrder): List<RecordingItemData> {
        val items = withContext(Dispatchers.IO) {
            getAudioFiles().sortedBy(sortOrder).map { RecordingItemData(it, RecorderType.AUDIO) }
        }
        return items
    }
```

Using Dispatchers.IO in File repository

Reply:
That'd work too, yes

Reply:
Using Dispatchers.IO in the file repository seems to have a noticeable performance improvement.

Reply:
Now the recording list view opens without any UI freezing